sway/desktop/render.c

```diff
@@ -1782,7 +1792,11 @@ void output_render(struct sway_output *output, struct timespec *when,
 
 		// Extend the damaged region
 		int expanded_size = effect_info.expanded_size;
-		if (expanded_size > 0) {
+		struct wlr_box damage_box = wlr_box_from_pixman_region32(damage);
+		if (expanded_size > 0
+				// Don't expand the damage any further if it's already
+				// larger than the monitor size
+				&& damage_box.width <= width && damage_box.height <= height) {
```

Comment:
we could instead check `damage->extents.width` / height, and wouldn't need the `wlr_box_from_pixman_region32()` function then :)

One more thing to consider: do we want to expand the region by the width if the height is greater than the monitor height, but the width has room to expand? or vice-versa?

Reply:
> we could instead check `damage->extents.width` / height, and wouldn't need the `wlr_box_from_pixman_region32()` function then :)

You mean something like this?:
```c
&& damage->extents.x2 - damage->extents.x1 <= width
&& damage->extents.y2 - damage->extents.y1 <= height
```
The `pixman_box32_t extents` extends struct sadly doesn't contain the width/height fields.

> One more thing to consider: do we want to expand the region by the width if the height is greater than the monitor height, but the width has room to expand? or vice-versa?

That is very true... I'm not sure if there is a wlroots function to only expand in one dimension though. Another idea would be to just check that both of the values are smaller than the `INT32_MAX` which would also solve the issue.

Reply:
Fixed in [eff5b81](https://github.com/WillPower3309/swayfx/pull/120/commits/eff5b8189400fb4384fdebd6a9a40837ba9a748f)